test/box-luatest/gh_6305_net_box_autocomplete_test.lua

```diff
@@ -0,0 +1,117 @@
+                        ['conn1:on_schema_reload('] = true,
+                        })
+
+    -- it should return all spaces and indexes
+    r = tabcomplete('conn1.space.s')
+    t.assert_equals(r, {['conn1.space.space1'] = true,
+                        })
+
+    r = tabcomplete('conn1.space.space1.ind')
+    t.assert_equals(r, {['conn1.space.space1.index'] = true,
+                        })
+
+    r = tabcomplete('conn1.space.space1.index.p')
+    t.assert_equals(r, {['conn1.space.space1.index.primary'] = true,
+                        })
+
+    r = tabcomplete('conn1.space.space1.index.primary:')
+    t.assert_equals(r, {['conn1.space.space1.index.primary:'] = true,
+                        ['conn1.space.space1.index.primary:max('] = true,
+                        ['conn1.space.space1.index.primary:select('] = true,
+                        ['conn1.space.space1.index.primary:update('] = true,
+                        ['conn1.space.space1.index.primary:delete('] = true,
+                        ['conn1.space.space1.index.primary:count('] = true,
+                        ['conn1.space.space1.index.primary:min('] = true,
+                        ['conn1.space.space1.index.primary:get('] = true,
+                        })
+
+    -- sreams are pretty the same
+    local s = c:new_stream()
+    rawset(_G, 'stream1', s)
+    r = tabcomplete('stream1.space.s')
+    t.assert_equals(r, {['stream1.space.space1'] = true,
+                        })
+
+    r = tabcomplete('stream1.space.space1.ind')
+    t.assert_equals(r, {['stream1.space.space1.index'] = true,
+                        })
+
+    r = tabcomplete('stream1.space.space1.index.p')
+    t.assert_equals(r, {['stream1.space.space1.index.primary'] = true,
+                        })
+
+    r = tabcomplete('stream1.space.space1.index.primary:')
+    t.assert_equals(r, {['stream1.space.space1.index.primary:'] = true,
+                        ['stream1.space.space1.index.primary:max('] = true,
+                        ['stream1.space.space1.index.primary:select('] = true,
+                        ['stream1.space.space1.index.primary:update('] = true,
+                        ['stream1.space.space1.index.primary:delete('] = true,
+                        ['stream1.space.space1.index.primary:count('] = true,
+                        ['stream1.space.space1.index.primary:min('] = true,
+                        ['stream1.space.space1.index.primary:get('] = true,
+                        })
+
+    -- futures
+    local f = c.space.space1:select({}, {is_async = true})
+    rawset(_G, 'future1', f)
+    r = tabcomplete('future1:')
+    t.assert_equals(r, {['future1:'] = true,
+                        ['future1:wait_result('] = true,
+                        ['future1:result('] = true,
+                        ['future1:discard('] = true,
+                        ['future1:pairs('] = true,
+                        ['future1:is_ready('] = true,
+                        })
+
+end
```

Comment:
Please split the patch in two:
1. The first one implements `__autocomplete` support for console and adds a test case that checks that autocomplete works fine for an arbitrary object that defines `__autcomplete` and/or `__index`.
2. The second one implements autocomplete support for net.box `future` and `stream` objects using the `__autocomplete` method introduced in the previous patch.